joeynmt/prediction.py

```diff
@@ -504,30 +542,28 @@ def write_to_file(output_path_set, hypotheses):
                     )
             else:
                 write_to_file("{}".format(output_path), all_hypotheses)
+                return output_path
         else:
-            # print to stdout
+            output_string=''
             for hyp in all_hypotheses:
-                print(hyp)
+                output_string+=hyp + "\n"
+            
+            return output_string
 
-    else:
-        # enter interactive mode
+    elif type_==2:
+        # single sentence is given
         batch_size = 1
         batch_type = "sentence"
-        while True:
```

Comment:
What happened to this interactive mode now? We'd like to keep this as a feature. Perhaps this requires breaking down the translate function into two separate functions.